File: C++/Q25.cpp
#include <iostream>
using namespace std;

int main() {
	long long n;
	cout << "Enter an integer: ";
	if (!(cin >> n)) {
		cout << "Invalid input." << endl;
		return 1;
	}
	long long t = n >= 0 ? n : -n;
	long long rev = 0;
	while (t > 0) {
		rev = rev * 10 + (t % 10);
		t /= 10;
	}
	if (n < 0) rev = -rev;
	cout << "Reversed number: " << rev << endl;
	return 0;
}

File: Java/Q25.java
import java.util.Scanner;

public class ReverseNumber {
    public static long reverse(long number) {
        long reversed = 0;
        while (number != 0) {
            long digit = number % 10;
            reversed = reversed * 10 + digit;
            number /= 10;
        }
        
        return reversed;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter a number to reverse: ");
        long num = sc.nextLong(); 
        long reversedNum = reverse(num);
        System.out.println("Reversed number is: " + reversedNum);

        sc.close();
    }
}